src/InfluxDB.LineProtocol/Payload/LineProtocolPoint.cs

```diff
@@ -37,40 +36,79 @@ public class LineProtocolPoint
+            writer.Measurement(Measurement);
 
             if (Tags != null)
             {
                 foreach (var t in Tags.OrderBy(t => t.Key))
                 {
-                    if (t.Value == null || t.Value == "")
+                    if (string.IsNullOrEmpty(t.Value))
                         continue;
 
-                    textWriter.Write(',');
-                    textWriter.Write(LineProtocolSyntax.EscapeName(t.Key));
-                    textWriter.Write('=');
-                    textWriter.Write(LineProtocolSyntax.EscapeName(t.Value));
+                    writer.Tag(t.Key, t.Value);
                 }
             }
 
-            var fieldDelim = ' ';
             foreach (var f in Fields)
             {
-                textWriter.Write(fieldDelim);
-                fieldDelim = ',';
-                textWriter.Write(LineProtocolSyntax.EscapeName(f.Key));
-                textWriter.Write('=');
-                textWriter.Write(LineProtocolSyntax.FormatValue(f.Value));
+                switch (f.Value)
+                {
+                    case sbyte value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case byte value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case short value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case ushort value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case int value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case uint value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case long value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case ulong value:
+                        if (value > long.MaxValue)
+                        {
+                            throw new InvalidOperationException("Influx cannot store a value larger than 9223372036854775807");
+                        }
+                        writer.Field(f.Key, (long)value);
+                        break;
+                    case float value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case double value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case decimal value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case bool value:
+                        writer.Field(f.Key, value);
+                        break;
+                    case TimeSpan value:
+                        writer.Field(f.Key, value.TotalMilliseconds);
+                        break;
```

Comment:
The point class accepts TimeSpan as a value which is not a LineProtocal data type. I kept this behaviour just in the `LineProtocolPoint` class rather than add it to the writer.